tests/dryrun/v3_install_test.go

```diff
@@ -0,0 +1,127 @@
+package dryrun
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/replicatedhq/embedded-cluster/pkg/dryrun"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestV3InstallHeadless_HappyPath(t *testing.T) {
+	licenseFile, configFile := setupV3HeadlessTest(t)
+
+	// Run installer command with headless flag and required arguments
+	err := runInstallerCmd(
+		"install",
+		"--headless",
+		"--target", "linux",
+		"--license", licenseFile,
+		"--config-values", configFile,
+		"--admin-console-password", "password123",
+		"--yes",
+	)
+
+	// Expect the command to fail with the specific error message
+	require.EqualError(t, err, "headless installation is not yet fully implemented - coming in a future release")
+
+	// PersistentPostRunE is not called when the command fails, so we need to dump the dryrun output manually
+	require.NoError(t, dryrun.Dump(), "fail to dump dryrun output")
+
+	t.Logf("Test passed: headless installation correctly returns not implemented error")
+}
+
+func TestV3InstallHeadless_Metrics(t *testing.T) {
```

Comment:
Hmm, should this just be part of the happy path test?

Reply:
My understanding from the testing strategy is that tests would be targeted per external dep. I assume we should count reporting as a dependency.

Reply:
Yeah you're right. I would say replicated.app is the external dependency here, but i guess reporting is the only functionality being tested so far. Maybe renaming the function to end with *ReplicatedApp might be better

Reply:
But the dryrun tests do not mock the api client, they are a level up from that mocking the metric events. So unless we were to refactor the dryrun framework to assert calls to replicated api could we leave this as is?